Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssBridge.ps1

```diff
@@ -0,0 +1,197 @@
+        }
+        else {
+            $correctType = $vss.Spec.Bridge -is $typeName
+            $vssBridgeTest += $correctType
+            if ($correctType) {
+                $comparingResult = Compare-Object -ReferenceObject $vss.Spec.Bridge.NicDevice -DifferenceObject $this.NicDevice
+                $areEqual = $null -eq $comparingResult
+                $vssBridgeTest += $areEqual
+                $vssBrdigeTest += ($vss.Spec.Bridge.Beacon.Interval -eq $this.BeaconInterval)
+                $vssBridgeTest += ($vss.Spec.Bridge.LinkDiscoveryProtocolConfig.Operation -eq $this.LinkDiscoveryProtocolOperation)
+                $vssBridgeTest += ($vss.Spec.Bridge.LinkDiscoveryProtocolConfig.Protocol -eq $this.LinkDiscoveryProtocolProtocol)
+            }
+        }
+        return ($vssBridgeTest -notcontains $false)
+    }
+
+    <#
+    .DESCRIPTION
+
+    Updates the Bridge configuration of the virtual switch.
+    #>
+    [void] UpdateVssBridge($vmHost) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $vssBridgeArgs = @{
+            Name = $this.VssName
+            NicDevice = $this.NicDevice
+            BeaconInterval = $this.BeaconInterval
+        }
+        if ($this.LinkDiscoveryProtocolProtocol -ne '') {
+            $vssBridgeArgs.Add('LinkDiscoveryProtocolProtocol', $this.LinkDiscoveryProtocolProtocol)
+            $vssBridgeArgs.Add('LinkDiscoveryProtocolOperation', $this.LinkDiscoveryProtocolOperation)
+        }
+        $vss = $this.GetVss()
+
+        if ($this.Ensure -eq 'Present') {
+            if ($this.Equals($vss)) {
+                return
+            }
+        }
+        else {
+            $vssBridgeArgs.NicDevice = @()
+            $vssBridgeArgs.BeaconInterval = 0
+        }
+        $vssBridgeArgs.Add('Operation', 'edit')
+
+        try {
+            Update-Network -NetworkSystem $this.vmHostNetworkSystem -VssBridgeConfig $vssBridgeArgs -ErrorAction Stop
+        }
+        catch {
+            Write-Error "The Virtual Switch Bridge Config could not be updated: $($_.Exception.Message)"
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Populates the result returned from the Get() method with the values of the Bridge settings of the Virtual Switch.
+    #>
+    [void] PopulateResult($vmHost, $vmHostVSSBridge) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $currentVss = $this.GetVss()
+
+        if ($null -ne $currentVss) {
+            $vmHostVSSBridge.VssName = $currentVss.Name
+            $vmHostVSSBridge.NicDevice = $currentVss.Spec.Bridge.NicDevice
+            $vmHostVSSBridge.BeaconInterval = $currentVss.Spec.Bridge.Beacon.Interval
+
+            if ($null -ne $currentVss.Spec.Bridge.linkDiscoveryProtocolConfig) {
+                $vmHostVSSBridge.LinkDiscoveryProtocolOperation = $currentVss.Spec.Bridge.LinkDiscoveryProtocolConfig.Operation
```

Comment:
ToString() the values from the server here.

Reply:
Updated